### PR TITLE
#2112. Add missing tests for mixins. Part 1

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class S {}
+typedef void Foo();
+
+class C1 = S with Function;
+//                ^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+class C2 = S with Foo;
```
Comment (eernstg) [inline Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart:27]: This one is actually a function type, and that error is still valid. So perhaps this test should be split into two libraries, where this one gets to keep the `@assertion` as it is.

File: Language/Mixins/Mixin_Classes/wrong_type_A06_t01.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+///     ⟨identifier⟩ ⟨typeParameters⟩? ‘=’ ⟨mixinApplication⟩ ‘;’
+/// ⟨mixinApplication⟩ ::= ⟨typeNotVoid⟩ ⟨mixins⟩ ⟨interfaces⟩?
+///
+/// It is a compile-time error if an element in the type list of the with clause
+/// of a mixin application is a type variable, a function type, a type alias
+/// that does not denote a class, an enumerated type, a deferred type, type
+/// dynamic, type void, or type FutureOr<T> for any T
+///
+/// @description Checks that it is a compile-time error if an element in the
+/// type list of the with clause of a mixin application is a type dynamic
```
Comment (eernstg) [inline Language/Mixins/Mixin_Classes/wrong_type_A06_t01.dart:16]:
```suggestion
/// type list of the with clause of a mixin application is the type `dynamic`
```

File: Language/Mixins/Mixin_Classes/wrong_type_A07_t01.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+///     ⟨identifier⟩ ⟨typeParameters⟩? ‘=’ ⟨mixinApplication⟩ ‘;’
+/// ⟨mixinApplication⟩ ::= ⟨typeNotVoid⟩ ⟨mixins⟩ ⟨interfaces⟩?
+///
+/// It is a compile-time error if an element in the type list of the with clause
+/// of a mixin application is a type variable, a function type, a type alias
+/// that does not denote a class, an enumerated type, a deferred type, type
+/// dynamic, type void, or type FutureOr<T> for any T
+///
+/// @description Checks that it is a compile-time error if an element in the
+/// type list of the with clause of a mixin application is a type void
```
Comment (eernstg) [inline Language/Mixins/Mixin_Classes/wrong_type_A07_t01.dart:16]:
```suggestion
/// type list of the with clause of a mixin application is the type `void`
```

File: Language/Mixins/Mixin_Classes/wrong_type_A08_t01.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+///     ⟨identifier⟩ ⟨typeParameters⟩? ‘=’ ⟨mixinApplication⟩ ‘;’
+/// ⟨mixinApplication⟩ ::= ⟨typeNotVoid⟩ ⟨mixins⟩ ⟨interfaces⟩?
+///
+/// It is a compile-time error if an element in the type list of the with clause
+/// of a mixin application is a type variable, a function type, a type alias
+/// that does not denote a class, an enumerated type, a deferred type, type
+/// dynamic, type void, or type FutureOr<T> for any T
+///
+/// @description Checks that it is a compile-time error if an element in the
+/// type list of the with clause of a mixin application is a type void
```
Comment (eernstg) [inline Language/Mixins/Mixin_Classes/wrong_type_A08_t01.dart:16]:
```suggestion
/// type list of the with clause of a mixin application is a type of the form
/// `FutureOr<T>` for some `T`.
```

**Review by eernstg — COMMENTED** [review]
Looks good! There are a few tests with a somewhat misleading description and a couple of typos.

File: Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+/// dynamic, type void, or type FutureOr<T> for any T
+///
+/// @description Checks that it is a compile-time error if an element in the
+/// type list of the with clause of a mixin application is a function type
+/// @author sgrekhov22@gmail.com
+
+class S {}
+typedef void Foo();
+
+class C1 = S with Function;
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart]: I removed `Function` from this test and added appropriate test to Class modifiers

File: Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart (hunk 1 above)
Comment (sgrekhov) [reply Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart]: Done

**Review by sgrekhov — COMMENTED** [review]
Thank you. Updated. Please take another look

File: Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply Language/Mixins/Mixin_Classes/wrong_type_A02_t01.dart]: Thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**